src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -205,12 +212,15 @@ protected virtual void OnCustomNodeRemoved(Guid functionId)
         /// <param name="info">
         ///     Custom node information data
         /// </param>
+        /// <param name="libraryServices">LibraryServices used for code block node initialization.</param>
+        /// <returns>Custom Node Instance</returns>
         public Function CreateCustomNodeInstance(
             Guid id, 
             string name,
             bool isTestMode, 
             CustomNodeDefinition def, 
-            CustomNodeInfo info)
+            CustomNodeInfo info,
+            Engine.LibraryServices libraryServices = null)
```

Comment:
is it really optional?

Reply:
I believe yes, only necessary for custom node workspace with CBN